libs/langchain/langchain/schema/runnable/base.py

```diff
@@ -67,6 +69,42 @@ class Runnable(Generic[Input, Output], ABC):
     """A Runnable is a unit of work that can be invoked, batched, streamed, or
     transformed."""
 
+    @property
+    def InputType(self) -> Type[Input]:
+        for cls in self.__class__.__orig_bases__:  # type: ignore[attr-defined]
```

Comment:
Is there an example of a runnable where the default implementation returns something? Most of the runnables seem to need to overload the input type

Reply:
see eg base Retriever